Sources/XMTPiOS/Conversations.swift

```diff
@@ -13,6 +17,20 @@ public actor Conversations {
 		self.client = client
 	}
 
+	public func groups() async throws -> [Group] {
+		try await client.v3Client.conversations().sync()
+
+		return try await client.v3Client.conversations().list(opts: .init(createdAfterNs: nil, createdBeforeNs: nil, limit: nil)).map { $0.fromFFI(client: client) }
+	}
+
+	public func newGroup(with addresses: [String]) async throws -> Group {
+		if addresses.isEmpty {
+			throw GroupError.emptyCreation
+		}
+
```

Comment:
I did a bunch of checks here in android that would probably be worth doing here as well

https://github.com/xmtp/xmtp-android/pull/160/files#diff-479fe53ececf1836cd910bfba9a721b20fce9fd212b204329abef95ff79429f9R90-R105

Reply:
I was relying on the errors bubbling up from libxmtp but I think you're right, having our own errors is nicer. Updated